x-pack/filebeat/docs/inputs/input-httpjson.asciidoc

```diff
@@ -1133,6 +1133,90 @@ request_url using file_name as 'file_2': \https://example.com/services/data/v1.0
 +
 Collect and make events from response in any format supported by httpjson for all calls.
 
++
+[[chain-step-replace_with]]
+[float]
+==== `chain[].step.replace_with`
+
+The `replace_with: "pattern,value"` clause is used to replace a pattern defined in the `request.url` with a given value. The value can be hard coded or can be 
+extracted from context variables like [`.last_response.*`, `.first_response.*`] etc. The `replace_with` clause can be used in combination with the `replace` clause 
+thus providing a lot of flexibility in the logic of chain requests.
+
+Example:
+
+- First call: \https://example.com/services/data/v1.0/exports
++
+<<response-json1-replace_with,response>>
+
+- Second call: \https://example.com/services/data/v1.0/$.exportId/files
++
+<<response-json2-replace_with,response>>
+
+["source","yaml",subs="attributes"]
+----
+filebeat.inputs:
+- type: httpjson
+  enabled: true
+  # first call
+  request.url: https://example.com/services/data/v1.0/exports
+  interval: 1h
+  chain:
+    # second call
+    - step:
+      request.url: https://example.com/services/data/v1.0/$.exportId/files
+      request.method: GET
+      replace_with: '$.exportId,first_response.body.exportId'
+----
+
+Example:
+
+- First call to fetch exportId
+
++
+request_url: \https://example.com/services/data/v1.0/exports
+
++
+response_json:
+
++
+[[response-json1-replace_with]]
+["source","json",subs="attributes"]
+----
+{
+    "exportId" : "2212"
+}
+----
+
+- Second call to fetch `file ids` using exportId from first call.
+
++
+request_url using exportId as '2212': \https://example.com/services/data/v1.0/2212/files
+
++
+response_json using exportId as '2212':
+
++
+[[response-json2-replace_with]]
+["source","json",subs="attributes"]
+----
+{
+    "files": [
+        {
+            "id": 1,
+        },
+        {
+            "id": 2,
+        },
+        {
+            "id": 3,
+        },
+    ]
+}
+----
+This behaviour of targeted pattern replacement in the url helps solve various use cases.
+
+NOTE: Replace patterns should not contain commas `","` in their definition.
```

Comment:
Doing this would remove the highlights. `` inside " " applies italics to text and does not highlight the text

Reply:
Single quotes? (aside: that's horrific)

Reply:
Single quotes also have the same behaviour : Italics & no highlights

Reply:
the best that can be done is a `"`,`"` .. which will italic the quotes ..suggesting the user that it's not a part of the text

Reply:
It looks like this is only possible by using a ZWJ between the backticks and the double quotes.

``"\U200D`,`\U200D"``.

Reply:
It's showing "\U200D`,`\U200D" in my local asciidoc preview window. It's not parsing "\U200D" . I'll just remove the symbol for comma and just highlight the text. It should be enough.

Reply:
Yeah that's a not a parsable bit of source. On a linux box it would be entered as ctrl-shift-u 200d, and it won't be visible in many editors; so it's not the way to go.